packages/abi-coder/src/Interface.ts

```diff
@@ -71,11 +71,11 @@ export class Interface<TAbi extends JsonAbi = JsonAbi> {
     const fragment =
       typeof functionFragment === 'string' ? this.getFunction(functionFragment) : functionFragment;
 
-    return fragment.decodeOutput(data);
+    return fragment.decodeOutput(data)[0];
   }
 
   decodeLog(data: BytesLike, logId: string): any {
-    const loggedType = this.jsonAbi.loggedTypes.find((type) => type.logId === logId);
+    const loggedType = this.jsonAbi.loggedTypes.find((type) => type.logId.toString() === logId);
```

Comment:
There's a bug in latest version after `logId` was changed to `string`.
Will `rb` become a `string` type? Why was the change needed?

Reply:
cc @nedsalk

Reply:
@DZakh the change is part of work on [#2313](https://github.com/FuelLabs/fuels-ts/issues/2313), specifically tied to https://github.com/FuelLabs/sway/issues/5953.

What is the bug you're referring to? Could you give me an example?

Reply:
I think that you might have tested this solution out with a `forc` version that's not `0.59.0`. After you build with it, you won't need this `type.logId.toString()` change because the `logId`s are strings.

Reply:
That's probably the case